app/code/Magento/Cms/Helper/Wysiwyg/Images.php

```diff
@@ -242,12 +242,16 @@ public function getCurrentPath()
                 }
             }
             try {
-                $currentDir = $this->_directory->getRelativePath($currentPath);
-                if (!$this->_directory->isExist($currentDir)) {
-                    $this->_directory->create($currentDir);
+                $currentTreePath = $this->_getRequest()->getParam('current_tree_path');
+                if ($currentTreePath) {
+                    $currentTreePath = $this->convertIdToPath($currentTreePath);
+                    $currentDir = $this->_directory->getRelativePath($currentTreePath);
+                    if (!$this->_directory->isExist($currentDir)) {
+                        $this->_directory->create($currentDir);
+                    }
                 }
             } catch (\Magento\Framework\Exception\FileSystemException $e) {
-                $message = __('The directory %1 is not writable by server.', $currentPath);
+                $message = __('The directory %1 is not writable by server.', $currentDir);
```

Comment:
Please escape `$currentDir`'s output

Reply:
Escape it for what exactly?

Reply:
@hostep If directory is not writable, script tags can get executed if passed as a variable value. I would recommend escapeHtml method.

Reply:
Ok, but this isn't the correct place for adding escaping, right? Should be done when that message is being outputted to the user (which is probably already happening), not in this Helper?
Or have you already tested this and found out that you can inject some executable code in there?

Reply:
@hostep You are right. Also, I don't think this can be exploited to affect any other user than the current logged in admin user.